### PR TITLE
geonkick: init at 2.3.3

### PR DESCRIPTION
**Pull request by magnetophon**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -3589,6 +3589,8 @@ in
 
   genimage = callPackage ../tools/filesystems/genimage { };
 
+  geonkick = callPackage ../applications/audio/geonkick {};
```
Comment (mmahut) [inline pkgs/top-level/all-packages.nix]: Can we please split these into separate commits?

Comment (magnetophon) [reply pkgs/top-level/all-packages.nix]: Geonkick is the only program in existence that uses redkite.
Does that change your opinion?
If not, I'll happily split them up.

Comment (magnetophon) [reply pkgs/top-level/all-packages.nix]: NM, I split them up anyway.
Thanks for the review!

**Comment by magnetophon** [comment]
@mmahut Could you have another look please?

**Comment by magnetophon** [comment]
I updated the version.

**Comment by magnetophon** [comment]
@mmahut @doronbehar @jtojnar Any comments on https://github.com/iurie-sw/geonkick/issues/87?

File: pkgs/applications/audio/geonkick/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  version = "2.2.0";
+
+  src = fetchFromGitLab {
+    owner = "iurie-sw";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0nb0q4pysgxpsm0zn5xgmb9xg2573pl8vx3b7q8pn8hr4pxrh8y2";
+  };
+
+  nativeBuildInputs = [ cmake pkgconfig ];
```
Comment (jtojnar) [inline pkgs/applications/audio/geonkick/default.nix:14]: In new expressions, `pkg-config` should be used instead of the `pkgconfig` alias.

File: pkgs/applications/audio/geonkick/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  buildInputs = [ redkite libsndfile rapidjson libjack2 lv2 libX11 cairo ];
+
+  enableParallelBuilding = true;
```
Comment (jtojnar) [inline pkgs/applications/audio/geonkick/default.nix:18]: `enableParallelBuilding` is enabled by default in meson and cmake packages.

File: pkgs/applications/audio/geonkick/default.nix (hunk 2 above)
Comment (magnetophon) [reply pkgs/applications/audio/geonkick/default.nix:14]: I keep hitting this one, I'll change it.  

When I made the package I searched for both options, found that pkgconfig is used way more, so I figured it was the correct one.

I guess nobody is willing to comb trough 7635 instances of ``pkgconfig`` and replace most, but not all, right?  Or is there another reason it is used more?

Comment (jtojnar) [reply pkgs/applications/audio/geonkick/default.nix:14]: Yeah, that is the case. Long long time ago nixpkgs did not allow dashes in attribute names so we omitted it. These days we are trying to move closer to upstream.

File: pkgs/applications/audio/geonkick/default.nix (hunk 3 above)
Comment (magnetophon) [reply pkgs/applications/audio/geonkick/default.nix]: Thanks, all done.

**Comment by magnetophon** [comment]
I don't undertsnd the error grahamcofborg-eval is giving.
Can someone explain?

**Comment by jtojnar** [comment]
Looks unrelated to this PR. Something is printing a warning when it is not supposed to.

**Comment by magnetophon** [comment]
Thanks.

**Comment by jtojnar** [comment]
This commit needs to be reverted: https://github.com/NixOS/nixpkgs/commit/7be315a842b3980a36b757bdfdc53365fa924e1f

**Comment by jtojnar** [comment]
@GrahamcOfBorg eval

File: pkgs/development/libraries/redkite/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1qg52m6pqilyq601qlwyp7dj921gis2j3vg7x5dpwrkb85iml71n";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ cairo ];
+
+  enableParallelBuilding = true;
```
Comment (jtojnar) [inline pkgs/development/libraries/redkite/default.nix:18]: `enableParallelBuilding` is enabled by default in meson and cmake packages.

File: pkgs/development/libraries/redkite/default.nix [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+  meta = {
+    homepage = "https://gitlab.com/iurie-sw/redkite";
+    description = "A small GUI toolkit";
+    license = stdenv.lib.licenses.gpl3;
```
Comment (jtojnar) [inline pkgs/development/libraries/redkite/default.nix:23]: Readme says GPL 3 but [source code](https://gitlab.com/iurie-sw/redkite/-/blob/7e15627dfd54c748129fe1cff9a4f83f034b1d9a/src/RkContainer.cpp#L9-11) says +. Needs clarification from upstream.

Comment (magnetophon) [reply pkgs/development/libraries/redkite/default.nix:23]: Good catch!
Do you always read the source of packages you review?

https://github.com/iurie-sw/redkite/issues/25

Comment (jtojnar) [reply pkgs/development/libraries/redkite/default.nix:23]: Only sometimes, mostly when checking for license or when something looks fishy.

Comment (magnetophon) [reply pkgs/development/libraries/redkite/default.nix]: All done.

I also changed the geonkick license to gpl3+, since that is what the code says.
I asked upstream to unify that too.

Comment (magnetophon) [reply pkgs/development/libraries/redkite/default.nix]: https://github.com/iurie-sw/redkite/issues/25#issuecomment-643269349
